src/fiatconnect/ReviewScreen.test.tsx

```diff
@@ -157,6 +160,52 @@ describe('ReviewScreen', () => {
       ])
     })
 
+    it('shows expired dialog when quote is expired', async () => {
+      const expireMs = -100
+      const props = getProps(CICOFlow.CashOut, false, CryptoType.cUSD, false, expireMs)
+      const quote = _.cloneDeep(mockFiatConnectQuotes[1]) as FiatConnectQuoteSuccess
+      quote.quote.guaranteedUntil = new Date(Date.now() + expireMs).toISOString()
+      const store = createMockStore({
+        fiatConnect: {
+          quotes: [quote],
+        },
+      })
+      store.dispatch = jest.fn()
+      const { getByTestId, queryByTestId } = render(
+        <Provider store={store}>
+          <FiatConnectReviewScreen {...props} />
+        </Provider>
+      )
+
+      expect(queryByTestId('expiredQuoteDialog')?.props.visible).toEqual(true)
+
+      await fireEvent.press(getByTestId('expiredQuoteDialog/PrimaryAction'))
+
+      expect(store.dispatch).toHaveBeenCalledWith(
+        refetchQuote({
+          flow: CICOFlow.CashOut,
+          quote: props.route.params.normalizedQuote,
+          fiatAccount: props.route.params.fiatAccount,
+        })
+      )
+    })
+    it('shows expired dialog when submitting expired quote', async () => {
+      jest.useRealTimers()
+      const expireMs = 100
+      const mockProps = getProps(CICOFlow.CashOut, false, CryptoType.cUSD, false, expireMs)
+      const { getByTestId, queryByTestId } = render(
+        <Provider store={store}>
+          <FiatConnectReviewScreen {...mockProps} />
+        </Provider>
+      )
+
+      expect(queryByTestId('expiredQuoteDialog')?.props.visible).toEqual(false)
+      await sleep(expireMs)
```

Comment:
nit: you can use jest fake timers instead of sleep https://jestjs.io/docs/timer-mocks#advance-timers-by-time

Reply:
That doesn't work because it only advances timers. It does not advance the global system time, which is what the quote's expiry time is compared against. I tried mocking the global Date object too, but that caused other tests to fail.

Reply:
Ah, I guess you can fake the time as well using [jest.setSystemTime](https://jestjs.io/docs/jest-object#jestsetsystemtimenow-number--date) on your specific time and then resetting it, but probably not worth it since it's just a 100ms wait.